pkg/apis/beat/v1beta1/beat_types.go

```diff
@@ -11,6 +11,10 @@ import (
 	commonv1 "github.com/elastic/cloud-on-k8s/pkg/apis/common/v1"
 )
 
+var (
+	KnownTypes = []string{"filebeat", "metricbeat", "heartbeat", "auditbeat", "journalbeat", "packetbeat"}
```

Comment:
Can we make this a set (`map[string]struct{}`) as it is searched through several times in the code below.

Reply:
I'd guess that with these 6 elements the slice will be smaller and faster, but I don't think it's worth an experiment and I've made the change :)